dts/bindings/video/video-interfaces.yaml

```diff
@@ -0,0 +1,133 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+# Common properties for video interface endpoints.
+
+include: base.yaml
+
+compatible: "zephyr,video-interfaces"
+
+description: |
+  A video pipeline usually consists of several devices, e.g. camera sensors, video
+  data receivers, video data processors, etc. Data interfaces on these devices are
+  described by their child 'port' nodes. Configuration of a port depends on other
+  devices in the pipeline and is described by 'endpoint' subnodes.
+
+  If a port can be configured to work with more than one remote device on the same
+  bus, an 'endpoint' child node must be provided for each of them. If more than one
+  port is present in a device node or there is more than one endpoint at a port, or
+  port node needs to be associated with a selected hardware interface, a common
+  scheme using '#address-cells', '#size-cells' and 'reg' properties is used.
+
+  All 'port' nodes can be grouped under an optional 'ports' node, which allows to
+  specify #address-cells, #size-cells properties independently for the 'port' and
+  'endpoint' nodes. For example:
+
+    video_device {
+      ...
+      ports {
+          #address-cells = <1>;
+          #size-cells = <0>;
+
+          port@0 {
+              ...
+              endpoint@0 { ... };
+              endpoint@1 { ... };
+          };
+          port@1 { ... };
+      };
+    };
+
+  Two 'endpoint' nodes must be linked with each other via their 'remote-endpoint'
+  phandles. However, Zephyr does not allow circular dependency, so direct phandle
+  references are currently not possible. A 'remote-endpoint-label' string is used
+  instead to be able to specify, at least, the label of the peer remote-endpoint.
+
+  This binding contains the most common properties needed to configure an endpoint
+  subnode for data exchange with other device. In most cases, properties at the
+  peer 'endpoint' node will be identical, however they might be different when
+  there is any signal modifications on the bus between two devices, e.g. there are
+  logic signal inverters on the lines.
+
+properties:
+  remote-endpoint-label:
```

Comment:
If I understand correctly, it is now possible to go through the various sensors and replace `source = <&source_dev>;` by the `remote-endpoint = <&source_dev>`?

Reply:
No, it is not possible as Zephyr does not support circular dependency as stated in the binding. So, replacing remote-endpoint phandle property by remote-endpoint-label (string type) is likely a workaround for this. 

I don't know if we can find a way to get the phandle of the remote-endpoint-label string by parsing the devicetree or not but at least, it will give us the name of the remote peer endpoint, if not, we don't know the endpoint will connect / interface with which endpoint.

Reply:
So for the MIPI/DVP DMA drivers:

```
port {
    endpoint {
        remote-endpoint = <&source_dev>;
    };
};
```

And on the image sensors:
```
port {
    endpoint {
        remote-endpoint-label = <&sink_dev>;
    };
};
```

When this is merged, I may open pull requests to convert the sensors to use the dts above?
Mostly changing the boards/shields bindings and the `DT_DEVICE_GET(DT_PHANDLE(...));` at the bottom of drivers AFAIU.

> we don't know the endpoint will connect / interface with which endpoint

We are in luck that sensors do not refer back to their sink currently !

Reply:
I am not sure if it's a good idead for a specific person to convert all the camera dts, shields to use the new bindings. Maybe it's better to let the owners if these camera dts, shields to do it gradually as they have boards to test it, and also there may be properties that they want to expose using the new bindings ? At least, for NXP, we prefer to do it ourselves.

Reply:
This sounds more reasonable. Thank you for clarifying.